pymc3/model.py

```diff
@@ -366,9 +367,17 @@ def __new__(cls, *args, **kwargs):
             instance._parent = cls.get_contexts()[-1]
         else:
             instance._parent = None
-        theano_config = kwargs.get('theano_config', None)
-        if theano_config is None or 'compute_test_value' not in theano_config:
-            theano_config = {'compute_test_value': 'raise'}
+        theano_config = kwargs.get('theano_config', {})
+        if 'compute_test_value' not in theano_config:
+            theano_config['compute_test_value'] = 'raise'
+        if theano_config['compute_test_value'] != theano.config.compute_test_value:
```

Comment:
But we should warn users if they set the value to something that doesn't compute a testval at all, and not when they change the flag. Usually, `theano.config.compute_test_value` will be `"off"` when this is called, right?